api/authorization.yml

```diff
@@ -1079,7 +1102,101 @@ paths:
           $ref: "#/components/responses/Unauthorized"
         default:
           $ref: "#/components/responses/ServerError"
+  /auth/users/{userId}/external/principals:
+      parameters:
+        - in: path
+          name: userId
+          required: true
+          schema:
+            type: string
+      get:
+        tags:
+          - auth
+          - experimental
+        parameters:
+          - $ref: "#/components/parameters/PaginationPrefix"
+          - $ref: "#/components/parameters/PaginationAfter"
+          - $ref: "#/components/parameters/PaginationAmount"
+        operationId: listUserExternalPrincipals
+        summary: list external principals for user
+        responses:
+          200:
+            description: external principals
+            content:
+              application/json:
+                schema:
+                  $ref: "#/components/schemas/ExternalPrincipalList"
+          401:
+            $ref: "#/components/responses/Unauthorized"
+          404:
+            $ref: "#/components/responses/NotFound"
+          default:
+            $ref: "#/components/responses/ServerError"
 
+  /auth/users/{userId}/external/principals/{principalId}:
+    parameters:
+      - in: path
+        name: userId
+        required: true
+        schema:
+          type: string
+      - in: path
+        name: principalId
+        required: true
+        schema:
+          type: string
+    post:
+      tags:
+        - auth
+        - experimental
+      operationId: createUserExternalPrincipal
+      summary: Create principal as external identity connected to lakeFS user
+      responses:
+        201:
+          description: external principal created successfully
+        401:
+          $ref: "#/components/responses/Unauthorized"
+        409:
+          $ref: "#/components/responses/Conflict"
```

Comment:
How?

Reply:
Elaborate?

Reply:
IIUC, you're asking in what case the Conflict error will be returned? 
If so then in case when an external principal is already attached to a user.